test/utils/Modifiers.t.sol

```diff
@@ -0,0 +1,529 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.22;
+
+import { ISablierLockup } from "src/core/interfaces/ISablierLockup.sol";
+
+import { Utils } from "./Utils.sol";
+
+abstract contract Modifiers is Utils {
+    /*//////////////////////////////////////////////////////////////////////////
+                                       COMMON
+    //////////////////////////////////////////////////////////////////////////*/
+
+    modifier givenCliffTimeNotInFuture(uint256 timestamp) {
+        vm.warp({ newTimestamp: timestamp });
+        _;
+    }
+
+    modifier givenEndTimeInFuture(uint256 timestamp) {
+        // Simulate the passage of time.
+        vm.warp({ newTimestamp: timestamp });
+        _;
+    }
+
+    modifier givenNFTExists() {
+        _;
+    }
+
+    modifier givenNotCanceledStream() {
+        _;
+    }
+
+    modifier givenNotNull() {
+        _;
+    }
+
+    modifier givenPreviousWithdrawal() {
+        _;
+    }
+
+    modifier givenRecipientAllowedToHook() {
+        _;
+    }
+
+    modifier givenStartTimeInPast() {
+        _;
+    }
+
+    /// @dev In LockupLinear, the streaming starts after the cliff time, whereas in LockupDynamic, the streaming starts
+    /// after the start time.
+    modifier givenSTREAMINGStatus(uint256 timestamp) {
+        // Warp to the future, after the stream's start time but before the stream's end time.
+        vm.warp({ newTimestamp: timestamp });
+        _;
+    }
+
+    modifier givenWarmStream() {
+        _;
+    }
+
+    modifier whenAssetContract() {
+        _;
+    }
+
+    modifier whenAssetERC20() {
+        _;
+    }
+
+    modifier whenAuthorizedCaller() {
+        _;
+    }
+
+    modifier whenCallerAdmin(address admin) {
```

Comment:
To avoid these, I would suggest the following:

1. Create a `Users` contract.
2. Move `Types` into `Users`
3. Initialize `users` variable in `Users` contract and create all of them
4. Inherit `Modifiers` from `Users`
5. Remove users setup from Base.

Simply use `users.admin` etc. in this contract.

Reply:
actually this was the initial version: https://github.com/sablier-labs/v2-core/commit/f5c973ce5f4d01771c015a29c5a9e4a49228bd1b

but i decided to change this, as passing the addresses in the modifiers, is more explicit IMO

Reply:
I see. Thanks for sharing the commit reference. 

> but i decided to change this, as passing the addresses in the modifiers, is more explicit IMO

It depends on the naming convention. When the modifier name is `whenCallerAdmin` and there is only one admin user declared as `users.admin`, there should not be any need to pass admin inside `whenCallerAdmin` when modifier name is explicit enough.

Why do you not agree with my suggestion? It would let us keep the modifier names as `whenCallerAdmin` and `whenCallerSender` without us requiring to pass input arguments.

Reply:
re introduced the previous version

Reply:
Nice thanks.